src/_P096_eInk.ino

```diff
@@ -0,0 +1,603 @@
+| rrf | r,<x>,<y>,<width>,<height>,<corner_radius>,<bordercolor>,<innercolor> | Draw a filled round rectangle |
+| px | px,<x>,<y>,<color> | Print a single pixel |
+  
+EPDCMD Subcommands:
+
+| EPD Subcommands | details | description |
+|-----|-----|-----|
+| clear | clear,<color> | Clear display |
+| deepsleep | deepsleep | Make screen go to sleep |
+| inv | inv,<value> | Invert the dispaly (value:0 normal display, 1 inverted display) |
+| rot | rot,<value> | Rotate display (value from 0 to 3 inclusive) |
+
+
+Examples:
+
+	Write Text :
+		http://<espeasy_ip>/control?cmd=epd,txtfull,0,0,HelloWorld
+	
+	Write Text another place:
+		http://<espeasy_ip>/control?cmd=epd,txtfull,100,40,HelloWorld
+
+	Write bigger Text :
+		http://<espeasy_ip>/control?cmd=epd,txtfull,0,0,3,HelloWorld
+
+	Write RED Text :
+		http://<espeasy_ip>/control?cmd=epd,txtfull,0,0,3,HelloWorld
+
+	Write RED Text (size is 1):
+		http://<espeasy_ip>/control?cmd=epd,txtfull,0,0,1,RED,HelloWorld
+
+	Write RED Text on YELLOW background (size is 1):
+		http://<espeasy_ip>/control?cmd=epd,txtfull,0,0,1,RED,YELLOW,HelloWorld
+		
+	Clear whole display
+		http://<espeasy_ip>/control?cmd=epdcmd,clear
+
+	Deepsleep screen
+		http://<espeasy_ip>/control?cmd=epdcmd,deepsleep
+
+*/
+
+//plugin dependency
+#include <LOLIN_EPD.h>
+#include <Adafruit_GFX.h>
+
+//declare functions for using default value parameters
+void Plugin_096_printText(const char *string, int X, int Y, unsigned int textSize = 1, unsigned short color = EPD_WHITE, unsigned short bkcolor = EPD_BLACK);
+
+//Define the default values for both ESP32/lolin32 and D1 Mini 
+#ifdef ESP32
+//for D32 Pro with EPD connector
+  #define EPD_CS 14
+  #define EPD_DC 27
+  #define EPD_RST 33  // can set to -1 and share with microcontroller Reset!
+  #define EPD_BUSY -1 // can set to -1 to not use a pin (will wait a fixed delay)
+#else
+  //for D1 Mini with shield connection
+  #define EPD_CS D0
+  #define EPD_DC D8
+  #define EPD_RST -1  // can set to -1 and share with microcontroller Reset!
+  #define EPD_BUSY -1 // can set to -1 to not use a pin (will wait a fixed delay)
+#endif
+
+
+//The setting structure
+struct Plugin_096_EPD_SettingStruct
+{
+  Plugin_096_EPD_SettingStruct()
+  : address_epd_cs(EPD_CS), address_epd_dc(EPD_DC), address_epd_rst(EPD_RST), address_epd_busy(EPD_BUSY), rotation(0), width(250), height(122)
+  {
+
+  }
+  byte address_epd_cs;
+  byte address_epd_dc;
+  byte address_epd_rst;
+  byte address_epd_busy;
+  byte rotation;
+  int width;
+  int height;
+} EPD_Settings;
+
+//The display pointer
+LOLIN_IL3897 *eInkScreen = NULL;
+byte plugin_096_sequence_in_progress = false;
+
+boolean Plugin_096(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+
+    case PLUGIN_DEVICE_ADD:
+      {
+        Device[++deviceCount].Number = PLUGIN_ID_096;
+        Device[deviceCount].Type = DEVICE_TYPE_I2C;
```

Comment:
It is not an I2C device, but SPI.
Currently we don't have a SPI type definition, but I guess we could need one.
Maybe you can add that type also?

Reply:
ok done